docs/configuration/configuration.md

```diff
@@ -1023,34 +1023,42 @@ discovery endpoints.
 The following meta labels are available on targets during relabeling:
 
 * `__meta_triton_groups`: the list of groups belonging to the target joined by a comma separator
-* `__meta_triton_machine_alias`: the alias of the target container
-* `__meta_triton_machine_brand`: the brand of the target container
-* `__meta_triton_machine_id`: the UUID of the target container
-* `__meta_triton_machine_image`: the target containers image type
-* `__meta_triton_server_id`: the server UUID for the target container
+* `__meta_triton_machine_alias`: the alias of the target virtual machine/global zone
+* `__meta_triton_machine_brand`: the brand of the target virtual machine or "gz" for global zones
+* `__meta_triton_machine_id`: the UUID of the target virtual machine/global zone
+* `__meta_triton_machine_image`: the target virtual machine image type
+* `__meta_triton_server_id`: the server UUID the target virtual machine is running on
+
+In case of global zones, the `groups`, `machine_image` and `server_id` labels are not applicable/set.
+Global zone alias (hostname) discovery depends on triton-cmon 1.7.0 or newer.
 
 ```yaml
 # The information to access the Triton discovery API.
 
-# The account to use for discovering new target containers.
+# The account to use for discovering new targets.
 account: <string>
 
-# The DNS suffix which should be applied to target containers.
+# The type of servers to discover, can be set to:
+# * "vm" to discover virtual machines (zones, bhyve vms) running on triton
+# * "gz" to discover global zones (head/compute nodes) making up the triton infrastructure
+[ server_type: <string> | default = "vm" ]
```

Comment:
Is server type really the term Triton uses for this?

Reply:
No, I just can't think of a better descriptor, perhaps something like discovery_type would be more fitting.

Reply:
role is what we use elsewhere, but I'm not quite sure that fits here.

Reply:
It seems the `role` as used in openstack is very similar and at least it's consistent within the prometheus config blocks. This would be a better name than server_type.

As to the values, I'm unhappy with the way triton is mixing their names, in the admin console it uses "server" and "virtual machine", but in the triton-cmon implementation, the roles are called 'container' and 'cn' (compute node) for the two discovery endpoints:
https://github.com/joyent/triton-cmon/blob/master/lib/endpoints/discover.js#L145

Neither solution matches reality as the vm/container instances are smartos zones but this name not seen in either location :-/
Although "container" is an overloaded term, I think it would be best to stick with the names used in triton-cmon as that is what the administrator will be looking at while configuring prometheus.

Reply:
Let's keep it consistent then, and put in a comment in the code as to which way we're going here. We can also mention the alternate names in the docs.

Reply:
`server_type` is changed to `role` and the docs are now a mirror of the openstack setup whith two lists and I've include all three variants of the names used.

I've added a comment block to the top of the refresh method:
https://github.com/prometheus/prometheus/pull/7250/files#diff-8f9f3d2f206c0f47ca80fb105f06f34bR153